docs/router/plugins.mdx

```diff
@@ -0,0 +1,265 @@
+---
+title: 'gRPC Plugins'
+description: 'Bridge any service into your GraphQL federation with secure, performant gRPC plugins simplifying architecture while enabling type-safe integration of REST, databases, and legacy systems'
+icon: "plug"
+---
+
+## Introduction
+
+The Cosmo Router supports extending its functionality with gRPC plugins, allowing you to implement subgraphs directly within the router process. This architecture provides significant performance benefits by reducing network hops while maintaining clean separation of concerns.
+
+<Frame type="color">
+  <div className="flex gap-3 items-center">
+    <div className="flex-shrink-0">
+      <img src="/images/router/grpc-plugins.png" width="100" />
+    </div>
+    <div>
+      <div className="font-semibold">Explore the demo repository</div>
+      <div className="text-sm">Check out our <a href="https://github.com/wundergraph/cosmo-plugin-demo" target="_blank" rel="noopener">cosmo-plugin-demo</a> repository for a working example of gRPC plugins in action.</div>
+    </div>
+  </div>
+</Frame>
+
+## What are gRPC Plugins?
+
+gRPC plugins are extensions that run as separate processes and communicate with the Cosmo Router over Google's Remote Procedure Call ([gRPC](https://grpc.io/)) protocol.
+
+Running as separate processes provides critical fault isolation - if a plugin crashes or encounters a fatal error, it won't bring down your router process.
+
+gRPC plugins implement the GraphQL resolvers while providing:
+
+<CardGroup cols={2}>
+  <Card title="Higher Performance" icon="bolt">
+    Reduce network hops by running resolvers inter-process with the router
+  </Card>
+  <Card title="Lower Latency" icon="gauge-high">
+    Reduce end-to-end request time with direct resolver calls
+  </Card>
+  <Card title="Resource Efficiency" icon="server">
+    Reduce infrastructure costs by consolidating services
+  </Card>
+  <Card title="Clean Architecture" icon="diagram-project">
+    Maintain separation of concerns with modular plugin design
+  </Card>
+</CardGroup>
```

Comment:
2 of these both read "lower latency" essentially, also not sure what "running resolvers inter-process with the router" means